packages/babel-traverse/src/path/introspection.js

```diff
@@ -157,7 +157,18 @@ export function isStatementOrBlock() {
  */
 
 export function referencesImport(moduleSource, importName) {
-  if (!this.isReferencedIdentifier()) return false;
+  if (!this.isReferencedIdentifier()) {
+    if (
+      (this.isMemberExpression() || this.isOptionalMemberExpression()) &&
+      (this.node.computed
+        ? t.isStringLiteral(this.node.property, { value: importName })
+        : this.node.property.name === importName)
+    ) {
+      return this.get("object").referencesImport(moduleSource, "*");
```

Comment:
Can you add an edge case on nested member expression?

```js
import * as ns from "foo";
ns["*"].bar

import { "*" as star } from "foo";
star.bar
```

I think `NodePath("ns["*"].bar").referencesImport("foo", "bar")` should return `false`, just like `NodePath("star.bar").referencesImport("foo", "bar")` does because `star` is not an imported namespace but a named import.

Reply:
Yeah this is sort of unfortunate API design, looking for a namespace import and an import called `'*'` is the same thing and this can't really be changed in a backwards-compatible way - if that problem wasn't there, this one wouldn't occur either. I guess we can work around it for the new code path at least, although the condition is kinda ugly and definitely feels like it shouldn't be there :sweat_smile: